### PR TITLE
document procedure to kill stale ros2 daemons

### PR DESCRIPTION
**Pull request by berndpfrommer**
This is related to issue #184

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,14 @@ ros2 run rmw_zenoh_cpp rmw_zenohd
 
 > Note: Without the zenoh router, nodes will not be able to discover each other since multicast discovery is disabled by default in the node's session config. Instead, nodes will receive discovery information about other peers via the zenoh router's gossip functionality. See more information on the session configs [below](#config).
 
+### Clear stale ros2 daemons
+```bash
+pkill -9 -f ros && ros2 daemon stop
+```
+Without this step the ROS2 CLI commands (e.g. ``ros2 node list``) may
```
Comment (Yadunund) [inline README.md]:
```suggestion
Without this step, ROS 2 CLI commands (e.g. `ros2 node list`) may
```

File: README.md [hunk 2]
```diff
@@ -47,6 +47,14 @@ ros2 run rmw_zenoh_cpp rmw_zenohd
...
+not work properly since they could still be using a ros2 daemon that
```
Comment (Yadunund) [inline README.md]:
```suggestion
not work properly since they would query ROS graph information from the ROS 2 daemon that
```

File: README.md [hunk 3]
```diff
@@ -47,6 +47,14 @@ ros2 run rmw_zenoh_cpp rmw_zenohd
...
+is running on a different RMW.
```
Comment (Yadunund) [inline README.md]:
```suggestion
may have been started with different a RMW.
```

File: README.md [hunk 4]
```diff
@@ -47,6 +47,14 @@ ros2 run rmw_zenoh_cpp rmw_zenohd
 
 > Note: Without the zenoh router, nodes will not be able to discover each other since multicast discovery is disabled by default in the node's session config. Instead, nodes will receive discovery information about other peers via the zenoh router's gossip functionality. See more information on the session configs [below](#config).
 
+### Clear stale ros2 daemons
```
Comment (Yadunund) [inline README.md]:
```suggestion
### Terminate ROS 2 daemon started with another RMW
```

**Review by Yadunund — CHANGES_REQUESTED** [review]
minor nitpicks

**Comment by Timple** [comment]
Maybe not the correct place to ask. But might ROS be improved such that the daemon is only spawned for rmw's that need it?

**Comment by berndpfrommer** [comment]
Accepted your suggested changes, LGTM.

**Review by Yadunund — APPROVED** [review]

**Comment by Yadunund** [comment]
> Maybe not the correct place to ask. But might ROS be improved such that the daemon is only spawned for rmw's that need it?

It may be worth having a discussion on https://github.com/ros2/ros2cli/tree/rolling. But imo even `rmw_zenoh` benefits from the daemon (which will be started automatically when there isn't one running and your run a `ros2 topic/node` command) since these graph queries can be made without initializing a new ROS context and hence waiting to connect with the Zenoh router; instead, we can query this information from the daemon node that has already connected to the router. Perhaps ` ros2 shutdown` command will be helpful to ensure all ROS 2 nodes and daemons are terminated before users switch to another RMW.

**Pull request merged**